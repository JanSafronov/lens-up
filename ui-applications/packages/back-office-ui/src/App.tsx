import './App.css'
import { DummyComponent } from '@lens-up/shared-components'

function App() {
  return (
    <>
      <h1 className="text-3xl font-bold underline">
        Hello world tailwind styles!
      </h1>
      <div>Back Office UI</div>
      <DummyComponent></DummyComponent>
    </>
  )
}

export default App
