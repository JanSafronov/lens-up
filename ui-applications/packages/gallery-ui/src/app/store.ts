import { configureStore } from '@reduxjs/toolkit'
import { galleryApi } from '../services/GalleryApi'

export const store = configureStore({
  reducer: {
    [galleryApi.reducerPath]: galleryApi.reducer
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware().concat(galleryApi.middleware),
})

export type RootState = ReturnType<typeof store.getState>
export type AppDispatch = typeof store.dispatch