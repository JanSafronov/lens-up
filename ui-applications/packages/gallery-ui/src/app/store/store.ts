import { configureStore } from '@reduxjs/toolkit'
import { galleryApi } from '../../services/GalleryApi'
import { galleryApiMiddleware } from './middlewares/GalleryApiMiddleware'

export const store = configureStore({
  reducer: {
    [galleryApi.reducerPath]: galleryApi.reducer
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware()
      .concat(galleryApi.middleware, galleryApiMiddleware),
})

export type RootState = ReturnType<typeof store.getState>
export type AppDispatch = typeof store.dispatch