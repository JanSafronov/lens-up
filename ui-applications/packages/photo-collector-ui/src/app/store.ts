import { configureStore } from '@reduxjs/toolkit'
import { photoCollectorApi } from '../services/PhotoCollectorApi'

export const store = configureStore({
  reducer: {
    [photoCollectorApi.reducerPath]: photoCollectorApi.reducer
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware().concat(photoCollectorApi.middleware),
})

export type RootState = ReturnType<typeof store.getState>
export type AppDispatch = typeof store.dispatch