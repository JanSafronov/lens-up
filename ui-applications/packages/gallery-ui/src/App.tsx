import './App.css'
import { DummyComponent } from '@lens-up/shared-components/src'

function App() {
  return (
    <>
      <div>Gallery UI</div>
      <DummyComponent></DummyComponent>
    </>
  )
}

export default App
