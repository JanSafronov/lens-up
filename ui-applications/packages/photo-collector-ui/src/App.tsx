import './App.css'
import { DummyComponent } from '@lens-up/shared-components/src'

function App() {
  return (
    <>
      <div>Photo Collector UI</div>
      <DummyComponent></DummyComponent>
    </>
  )
}

export default App
