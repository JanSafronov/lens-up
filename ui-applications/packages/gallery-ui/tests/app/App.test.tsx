import renderer from 'react-test-renderer';
import {expect, it} from '@jest/globals';
import App from '../src/App';
import { createAsyncThunk } from '@reduxjs/toolkit';
import { create } from './app/store.test';

it('Middleware must shows action rejection', () => {
    const { next, invoke } = create();
    //const action = { type: 'rejected' };
    const action = createAsyncThunk("test/rejected", async (state: string, { rejectWithValue }) => {
        if (state != "accepted")
        return rejectWithValue('No user found');
      });
    invoke(action);
    expect(next).not.toHaveBeenCalledWith(action);
  });

it('App must render', () => {
    const component = renderer.create(
        <App />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });