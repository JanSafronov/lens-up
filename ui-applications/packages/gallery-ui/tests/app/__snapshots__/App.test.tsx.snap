// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App must be rendering 1`] = `
{
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "logTestingPlaygroundURL": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
}
`;

exports[`App must render according to snapshot 1`] = `
<DocumentFragment>
  <div
    class="min-h-screen bg-black"
  >
    <div
      class="container mx-auto flex flex-wrap flex-col md:flex-row justify-center h-screen items-center"
    >
      <div>
        <img
          alt="LensUp logo."
          height="147"
          src="/@fs/C:/Users/safronov/source/lens-up/ui-applications/packages/shared-components/src/images/lens-up-logo.png"
          width="379"
        />
        <form
          class="px-7 py-4 bg-black rounded-lg leading-none border-double border-4 border-white shadow-lg max-w-sm mx-auto"
        >
          <div
            class="mb-5"
          >
            <div>
              <label
                class="block mb-2 text-sm font-medium text-gray-50 dark:text-white"
                for=":r0:"
              >
                Join to your gallery:
              </label>
              <input
                class="bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-lg focus:ring-blue-500 focus:border-blue-500 block w-full p-2.5 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500"
                id=":r0:"
                placeholder="Enter code"
                required=""
                type="number"
                value=""
              />
            </div>
          </div>
          <div
            class="flex items-center justify-center"
          >
            <div
              class="text-center relative group"
            >
              <button
                class="text-white bg-blue-700 hover:bg-blue-800 focus:ring-4 focus:outline-none focus:ring-blue-300 font-medium rounded-full text-sm w-full sm:w-auto px-5 py-2.5 text-center dark:bg-blue-600 dark:hover:bg-blue-700 dark:focus:ring-blue-800 inline-flex items-center justify-center"
                type="submit"
              >
                Join
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Middleware must shows action rejection 1`] = `[Function]`;
